### PR TITLE
Remove legacy resource references in multicluster application docs

### PR DESCRIPTION
**Pull request by desagar**

### CODE REVIEW

**Review by kminder — APPROVED** [review]
Approved.

**Review by mgianatagh — APPROVED** [review]

File: content/en/docs/applications/multicluster/_index.md [hunk 1]
```diff
@@ -39,20 +39,17 @@ installation profile. A managed cluster has the following additional characteris
...
+Verrazzano includes a [MultiClusterApplicationConfiguration]({{< relref "/docs/reference/api/multicluster/multiclusterapplicationconfiguration" >}})
+resource definition for applications that may be targeted for placement in one or more clusters. 
+
+- A MultiClusterApplicationConfiguration is a wrapper for an ApplicationConfiguration, and additionally allows the
+  `placement` of the underlying resource to be specified as a list of names of the clusters in which the
+  ApplicationConfiguration must be placed.
+- MultiClusterApplicationConfiguration resources, along with their associated Component and Secret resources, are
+  created in the admin cluster, in a namespace that is part of a VerrazzanoProject, and targeted for `placement`
+  in either the local admin cluster or a remote managed cluster.
+- A multicluster application is said to be part of a VerrazzanoProject if it is in a namespace that is governed
```
Comment (rosemarymarano) [inline content/en/docs/applications/multicluster/_index.md:52]: is said to be part -> is considered part

**Review by rosemarymarano — APPROVED** [review]
One minor edit, otherwise fine.

File: content/en/docs/applications/multicluster/_index.md (hunk 1 above)
Comment (desagar) [reply content/en/docs/applications/multicluster/_index.md]: done

**Pull request merged**